pkg/backup/backup.go

```diff
@@ -421,5 +461,7 @@ func (*realItemBackupper) backupItem(ctx *backupContext, item map[string]interfa
 		return errors.WithStack(err)
 	}
 
+	ctx.backedUpItems[key] = struct{}{}
```

Comment:
Move this up to just below where you check if we've already backed it up, so we record it asap

Reply:
I left it down here because I didn't want to record it as backed up unless it actually succeeded. WDYT?

Reply:
My gut says that trying a second time if the initial backup failed would likely encounter the same error?